src/proto/h1/io.rs

```diff
@@ -168,12 +168,9 @@ where
                     }
                 }
             }
-            match ready!(self.poll_read_from_io(cx)).map_err(crate::Error::new_io)? {
-                0 => {
-                    trace!("parse eof");
-                    return Poll::Ready(Err(crate::Error::new_incomplete()));
-                }
-                _ => {}
+            if let 0 = ready!(self.poll_read_from_io(cx)).map_err(crate::Error::new_io)? {
```

Comment:
`if let 0 =` reads odd to me. Should just be `if $expr == 0`?

Reply:
copy that, will update

Reply:
done.